### ISSUE TITLE
mirrored mode doesn't adapt to changing window size (e.g. fullscreen)

### ISSUE THREAD

**Issue by grrrwaaa** [num3ric/Cinder-OculusRift#5]
Hi, this block is awesome, worked right away. One thing:

If I enter fullscreen, the mirror texture is blitted into the backbuffer at the same resolution as the original window, occupying a small fraction of the actual window. 

I worked around this by modifying the constructor to use the default display dimensions for the mirror texture, 

initializeMirrorTexture(app::getWindow()->getDisplay()->getSize());

and the blit code to respect this difference:

GLint src_w = mMirrorTexture->OGL.Header.TextureSize.w;
        GLint src_h = mMirrorTexture->OGL.Header.TextureSize.h;
        const ivec2 windowsize = app::getWindowSize();
        GLint dst_w = windowsize.x;
        GLint dst_h = windowsize.y;
        glBlitFramebuffer( 0, src_h, src_w, 0,
            0, 0, dst_w, dst_h,
            GL_COLOR_BUFFER_BIT, GL_LINEAR );

Seems to work OK, but I'm not sure if this is the ideal solution.

BTW I would also really like to be able to blit the left or right eye texture, because sometimes the mirror screen is better without the Rift distortions etc. applied. Is there a way to do this?

Thanx

**Comment by num3ric** [comment]
Thanks for reporting. The above commit should address this problem. Let me know if that works.

As for the mono mirror texture, I don't think there is a simple solution that _doesn't increase rendering costs_ (i.e. increasing one eye's viewport or rendering the scene a third time). The current mirror texture is provided directly by the SDK.

**Comment by num3ric** [comment]
Note that there is another issue with the mirror texture atm: disabling mirroring will cause flickering in the app's window.

I won't bother fixing it now though, not until they fix the WM_PAINT bug.

**Comment by grrrwaaa** [comment]
Awesome -- I'll check it once I get back to the studio where the DK2 is.

Yes also noted the flickering, but I thought it was something I was doing wrong -- I'm still getting to know Cinder.

About the other texture access, I didn't mean a mono render of the mirror, I mean the texture of the FBO that the scene is rendered into -- would prefer the option to have this undistorted view available on the mirror window rather than the DK2-distorted texture. I mean, the mRenderBuffer->TextureSet texture, rather than the mMirrorFBO texture.  

Maybe it can be done already, by blitting from the current draw buffer just before the ScopedRiftBuffer unbinds (before submitFrame)?

**Comment by num3ric** [comment]
@grrrwaaa Ah, I see. Have you seen this supported elsewhere (other than Google Cardboard)? I haven't seen it option in the SDK samples, nor in UE4. I would prefer "obeying" the SDK, and only using the ovr_CreateMirrorTextureGL.

**Comment by num3ric** [comment]
@grrrwaaa Got it. And deleted my previous message: looks like this should be allowed by the SDK, at least on the D3D11 with the "Mirror Variants" sample. Maybe your approach would work.
